zerver/views/auth.py

```diff
@@ -424,22 +424,19 @@ def authenticate_remote_user(realm, email_address):
                                 return_data=return_data)
     return user_profile, return_data
 
-def log_into_subdomain(request):
-    # type: (HttpRequest) -> HttpResponse
+_subdomain_token_salt = 'zerver.views.auth.log_into_subdomain'
+
+def log_into_subdomain(request, token):
+    # type: (HttpRequest, Text) -> HttpResponse
     try:
-        # Discard state if older than 15 seconds
-        state = request.get_signed_cookie('subdomain.signature',
-                                          salt='zerver.views.auth',
-                                          max_age=15)
-    except KeyError:
-        logging.warning('Missing subdomain signature cookie.')
-        return HttpResponse(status=400)
+        data = signing.loads(token, salt=_subdomain_token_salt, max_age=15)
+    except signing.SignatureExpired as e:
+        logging.warning('Subdomain cookie: {}'.format(e))
```

Comment:
Shouldn't this return an error HttpResponse?

Reply:
Good catch, thanks!